editor/components/post-preview-button/index.js

```diff
@@ -123,14 +124,25 @@ export class PostPreviewButton extends Component {
 	}
 }
 
-export default connect(
-	( state ) => ( {
-		postId: state.currentPost.id,
-		link: getEditedPostPreviewLink( state ),
-		isDirty: isEditedPostDirty( state ),
-		isNew: isEditedPostNew( state ),
-		isSaveable: isEditedPostSaveable( state ),
-		modified: getEditedPostAttribute( state, 'modified' ),
+export default compose(
+	withSelect( ( select ) => {
+		const { getEditedPostAttribute } = select( 'core/editor' );
+		const { getPostType } = select( 'core' );
+		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		return {
+			isPreviewable: postType && postType.previewable,
+		};
 	} ),
-	{ autosave }
+	ifCondition( ( { isPreviewable } ) => isPreviewable ),
+	connect(
```

Comment:
Everything in the `connect()` should be moved to the `withSelect()`. This also avoids having to import `getEditedPostAttribute()` twice.

Reply:
I'm not sure I follow. Can you clarify what you mean by this, or point me to an example?

Reply:
Sure, check out the `withSelect()` usage in the [post title component](https://github.com/WordPress/gutenberg/blob/master/editor/components/post-title/index.js#L127-L136).

Anything that's coming from the editor selectors should be returned from the `withSelect()` function. For example, the current `modified: getStateEditedPostAttribute( state, 'modified' )` in `connect()` would be moved to the object returned from the `withSelect()` function, so it becomes:

```js
return {
	isPreviewable: get( postType, 'viewable', false ),
	modified: getEditedPostAttribute( 'modified' ),
};
```

The same thing can be repeated for all of the items in the `connect()`.